### ISSUE TITLE
z.detach() in trainer step() blocks end-to-end reconstruction gradient

### ISSUE THREAD

**Issue by epignatelli** [epignatelli/latent-option-models-code#1]
## Bug

Both `LAMTrainer.step` and `LOMTrainer.step` detach the latent vector before passing it to the dynamics model:

**LAMTrainer** (`training.py` line 326):
\`\`\`python
logits = self.models["dynamics"](history, z.detach())
\`\`\`

**LOMTrainer** (`training.py` lines 394, 401, 402):
\`\`\`python
z_act, vq_act, _ = self.models["action_lam"](history, next_frame, z_opt.detach())
lam_logits = self.models["lam_dynamics"](history, z_act.detach())
lom_logits = self.models["lom_dynamics"](history, z_act.detach(), option_code=z_opt.detach(), ...)
\`\`\`

## Impact

The dynamics reconstruction loss cannot backpropagate through the LAM/option encoders. The encoders train only via VQ commitment loss, not end-to-end. This severely limits what the latent space learns.

## Fix

Remove all `.detach()` calls on `z`, `z_act`, and `z_opt` before passing to dynamics models.
